core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -93,14 +93,15 @@ oppia.factory('responsesService', [
               function() {
                 explorationStatesService.updateSolutionValidity(
                   currentStateName, true);
-                alertsService.addInfoMessage(
-                  'The solution is now valid!');
+                explorationWarningsService.updateWarnings();
+                alertsService.addInfoMessage('The solution is now valid!');
               },
               function() {
                 explorationStatesService.updateSolutionValidity(
                   currentStateName, false);
+                explorationWarningsService.updateWarnings();
                 alertsService.addInfoMessage(
-                  'That solution does not lead to the next state!');
+                  'The current solution is not valid for this rule.');
```

Comment:
Do the validity check of the previous solution here (probably somewhere above), and only show "The current solution is no longer valid." if the solution was fine before and is now no longer fine.